analysis/org.eclipse.tracecompass.analysis.timing.core/src/org/eclipse/tracecompass/internal/analysis/timing/core/segmentstore/SegmentStoreStatisticsDataProvider.java

```diff
@@ -116,17 +117,26 @@ public SegmentStoreStatisticsDataProvider(ITmfTrace trace, IStatisticsAnalysis p
         fModule = provider instanceof IAnalysisModule ? (IAnalysisModule) provider : null;
     }
 
+    @SuppressWarnings("null")
     @Override
     public TmfModelResponse<TmfTreeModel<SegmentStoreStatisticsModel>> fetchTree(Map<String, Object> fetchParameters, @Nullable IProgressMonitor monitor) {
         IAnalysisModule module = fModule;
+        boolean success = true;
         if (module != null) {
             if (monitor != null) {
-                module.waitForCompletion(monitor);
+                success = module.waitForCompletion(monitor);
                 if (monitor.isCanceled()) {
```

Comment:
If there is failure but also monitor canceled, should the failure override the cancel? i.e. should we return cancelled only if there is no failure cause?